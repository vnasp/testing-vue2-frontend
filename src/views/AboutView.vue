<template>
  <div class="about">
    <h1>Sobre</h1>
  </div>
</template>

<script>
export default {
  name: 'AboutView',
  data: function () {
    return {};
  }
}
</script>
