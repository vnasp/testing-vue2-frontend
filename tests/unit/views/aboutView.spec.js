import { shallowMount } from '@vue/test-utils'
import { createRouter, createWebHistory } from 'vue-router'
import AboutView from '@/views/AboutView.vue'

describe('AboutView', () => { 
    test('Probando la existencia de la vista AboutView', async () => {
        const router = createRouter({
            history: createWebHistory(),
            routes: [{
                path: '/about',
                name: 'AboutView',
                component: AboutView
            }],
        })
        router.push('/about')
        await router.isReady()

        const wrapper = shallowMount(AboutView, {
            global: {
                plugins: [router]
            }
        })
        expect(wrapper.findComponent(AboutView).exists()).toBe(true)
    })  

    test('Crea / valida un snapshot de la vista AboutView', async () => {
        const wrapper = shallowMount(AboutView)
        expect(wrapper.html()).toMatchSnapshot()
    })  
})
