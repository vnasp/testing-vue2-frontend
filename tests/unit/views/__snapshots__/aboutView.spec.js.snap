// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutView Crea / valida un snapshot de la vista AboutView 1`] = `
<div class="about">
  <h1>Sobre</h1>
</div>
`;
